src/modules/Radio/Radio.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import uuid from "uuid/v4";
+import {
+  StandardProps,
+  WithStylesProps,
+  ThemeStyleSheetFactory
+} from "../../types";
+import { withStyles } from "../../base";
+
+interface IProps extends StandardProps<"input"> {
+  hasError?: boolean;
+  isDisabled?: boolean;
+  label: string;
+  name?: string;
+  selectedValue?: string;
+  value: string;
+  onValueSelect?: (value: string) => void;
+}
+
+const styleSheet: ThemeStyleSheetFactory = theme => ({
+  input: {
+    color: theme.colors.primary,
+    cursor: "pointer",
+    ":hover": {},
+    ":active": {},
+    "@selectors": {
+      "&.checked": {},
+      "&.disabled": {
+        cursor: "default"
+      },
+      "&.error": {}
+    }
+  },
+  label: {
+    cursor: "pointer",
+    ":hover": {},
+    ":active": {},
+    "@selectors": {
+      "&.checked": {},
+      "&.disabled": {
+        cursor: "default"
+      },
+      "&.error": {
+        color: theme.colors.alert.alert50
+      }
+    }
+  },
+  wrapper: {
+    display: "flex",
+    alignItems: "center",
+    "@selectors": {
+      "&.checked": {},
+      "&.disabled": {},
+      "&.error": {
+        color: theme.colors.alert.alert50
+      }
+    }
+  }
+});
+
+class Radio extends React.PureComponent<IProps & WithStylesProps> {
+  // eslint-disable-next-line react/state-in-constructor
+  state = {
+    id: uuid()
+  };
+
+  handleChange() {
+    const { isDisabled, value, onValueSelect } = this.props;
+
+    if (!isDisabled && onValueSelect) {
+      onValueSelect(value);
+    }
+  }
+
+  render() {
+    const {
+      cx,
+      hasError,
+      isDisabled,
+      label,
+      name,
+      selectedValue,
+      styles,
+      value,
+      wrappedRef
+    } = this.props;
+    const { id } = this.state;
+    const isChecked = selectedValue === value;
+
+    return (
+      <div className={cx(styles.wrapper)}>
+        <input
+          checked={isChecked}
+          className={cx(
+            styles.input,
+            isChecked && "checked",
+            isDisabled && "disabled",
+            hasError && "error"
+          )}
+          id={id}
+          name={name}
+          ref={wrappedRef}
+          type="radio"
+          value={value}
+          onChange={this.handleChange.bind(this)}
+        />
```

Comment:
you can use {...props} after "removing" all the custom props on the beginning like (label, ...props)